pkgs/controllers/certificaterequest.go

```diff
@@ -135,6 +139,15 @@ func (r *CertificateRequestController) Reconcile(ctx context.Context, cr *certma
 	}
 
 	pem, err := p.Sign(ctx, cr)
+
+	var apiError cfapi.APIError
```

Comment:
Should this be `*cfapi.APIError` b/c the error returned by `Sign` will be a `*cfapi.APIError`? https://github.com/cloudflare/origin-ca-issuer/blob/9cbc0d54b1c5ca7971dabe114b3891478a2e7c90/internal/cfapi/cfapi.go#L120

Reply:
https://goplay.space/#qKH9IobZ_4f

Reply:
I was debating on if `APIError` or `*APIError` should implement the `error` interface, on account that the pointer receiver made this `error.As` check look bad. The original PR had partially switched to `APIError` but missed this one case. I've added more tests to verify the error type, and reverted back to `*APIError` after finding the Go blog posts also use the pointer receiver.